tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -30,6 +44,183 @@
+
+        if not self.skip_dom_polling_handle:
+            logging.info("Restore DOM polling to {} after sfp reset for {}".format(self.orig_dom_polling_value,
+                                                                                   self.logical_intf))
+            if not self.orig_dom_polling_value:
+                restore_dom_result = self.duthost.command(self.db_cmd_dom_polling_clear)
+            else:
+                restore_dom_result = self.duthost.command(db_cmd_dom_polling.format(self.namespace_cmd_opt,
+                                                                                    "HSET",
+                                                                                    self.logical_intf,
+                                                                                    self.orig_dom_polling_value))
+            assert restore_dom_result["rc"] == 0, "Restore DOM polling failed for {}".format(self.logical_intf)
+
+
+class DisablePhysicalInterface:
+    """
+    Context manager to disable the given physical interface (as wells as its
+    logical interfaces if needed) and restore afterwards.
+
+    Disable/enable port includes:
+        * Disable/enable DOM polling
+        * Shutdown/startup port
+        * Check/restore lpmode
+    """
+    def __init__(self, duthost, enum_frontend_asic_index, phy_intf, logical_intfs_list):
+        self.duthost = duthost
+        self.phy_intf = phy_intf
+        self.original_lpmode_state = None
+        self.wait_after_dom_config = 1
+        self.wait_after_lpmode_set = 1
+        self.logical_intfs_list = logical_intfs_list
+        self.namespace_cmd_opt = get_namespace_cmd_option(duthost,
+                                                          enum_frontend_asic_index)
+        self.skip_lpmode_handle_global = self.skip_lpmode_handle_for_physical_interface()
+        self.logical_intf_disablers = \
+            [LogicalInterfaceDisabler(duthost=duthost,
+                                      enum_frontend_asic_index=enum_frontend_asic_index,
+                                      logical_intf=logical_intf,
+                                      phy_intf=phy_intf,
+                                      skip_dom_polling_handle=(i != 0),
+                                      skip_lpmode_handle=(self.skip_lpmode_handle_global or
+                                                          i != 0))
+             for i, logical_intf in enumerate(logical_intfs_list)]
+
+    def skip_lpmode_handle_for_physical_interface(self):
+        cmd = "sonic-db-cli {} STATE_DB HGETALL 'TRANSCEIVER_INFO|{}'".format(self.namespace_cmd_opt,
+                                                                              self.logical_intfs_list[0])
+        xcvr_info_output = self.duthost.command(cmd)["stdout"]
+        is_cmis = "cmis_rev" in xcvr_info_output
+        is_power_class_1 = "Power Class 1" in xcvr_info_output
+
+        if is_cmis or is_power_class_1:
```

Comment:
could you please add comment about why cmis is skipped?

Reply:
@longhuan-cisco Let's not skip lpmode toggling for CMIS modules.
Based on my analysis, the toggling is needed since some platforms may not have LowPwrRequestHW signal ASSERTED. Hence, triggering `sfputil reset` will not cause any change to Module State for such platforms and will eventually require user to toggle the lpmode manually.

Reply:
> could you please add comment about why cmis is skipped?

added

Reply:
> @longhuan-cisco Let's not skip lpmode toggling for CMIS modules. Based on my analysis, the toggling is needed since some platforms may not have LowPwrRequestHW signal ASSERTED. Hence, triggering `sfputil reset` will not cause any change to Module State for such platforms and will eventually require user to toggle the lpmode manually.

@mihirpat1 

<img width="1008" alt="image" src="https://github.com/user-attachments/assets/23c4bf66-a66b-45cc-ac67-5c457688fb04">
Doesn't cmis_mgr always turns CMIS module to high power (via `cmis_api.set_lpmode(False)`)  in CMIS state machine when starting up the interface? And set_lpmode forces transition from LowPwr to HighPower state under SW control regardless of LowPwrRequestHW.

https://github.com/sonic-net/sonic-platform-common/blob/daeed65100c9250a704de223a771fdc7416a90f0/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L1089

thus, I didn't follow why CMIS module still needs manual toggle (`sfputil lpmode`) after reset.